### ISSUE TITLE
Win Condition not Satisfied

### ISSUE THREAD

**Issue by ninjawithsushi** [eecs183/Connect4#1]
Win condition is not being triggered, the game is in a pseudo-infinite loop

Press 1 for terminal and 2 for graphics: 1
Enter L to load a game, or P to play: p
Player 1, enter your name: bob

Player 2, enter your name: alice

Enter: the column to move, q to quit, s to save
bob enter your move: 1

Enter: the column to move, q to quit, s to save
alice enter your move: 2

Enter: the column to move, q to quit, s to save
bob enter your move: 1

Enter: the column to move, q to quit, s to save
alice enter your move: 2

Enter: the column to move, q to quit, s to save
bob enter your move: 1

Enter: the column to move, q to quit, s to save
alice enter your move: 2

Enter: the column to move, q to quit, s to save
bob enter your move: 1

Enter: the column to move, q to quit, s to save
alice enter your move:

bob has clearly dropped four checkers into col 1, win condition should have been triggerd

**Comment by cherinac** [comment]
Hi Alex, 
when we are testing our group's connect4.cpp, it has the win condition. As the test case came out, it shows that our program terminated early. Can you help us know which part of the code has caused this condition?
the screen shot is what I had on my computer when I have four in a row with the same code submitted to github.
<img width="326" alt="screen shot 2016-12-08 at 8 06 53 pm" src="https://cloud.githubusercontent.com/assets/23159961/21033892/f385bf86-bd81-11e6-8750-5b369bb0ceae.png">

**Comment by ninjawithsushi** [comment]
Hello cherinac, 
Thank you for your prompt reply to the issue bug I have on win-condition. I have used the auto-grader to debug the issue and it has terminated early because of pre-compilation error. It seems that board.cpp uses STL library containers (i.e. std::vector) in the implementation of the FEN string and autograder complains that std::vector has not been included in board.cpp

I have fixed it simply by including vector in board.cpp.

Admittedly, I had planned to use Github's bug-tracker as points of work as we continue to debug the core, but I forgot that the repo creators would get notifications for the bugs that we as a group would post. At least now we can receive feedback or helpful notifications if we continue to use it. 

Thank you, again.

-Alex
